e2e/testcases/helm_sync_test.go

```diff
@@ -205,47 +177,58 @@ service:
 		}
 	  }`)
 
-	err := nt.WatchForAllSyncs(nomostest.WithRootSha1Func(helmChartVersion("15.2.35")),
-		nomostest.WithSyncDirectoryMap(map[types.NamespacedName]string{nomostest.DefaultRootRepoNamespacedName: "wordpress"}))
-	if err != nil {
+	nt.WaitForRootSyncStalledError(rs.Namespace, rs.Name, "Validation", "KNV1061: RootSyncs must reference valid ConfigMaps in spec.helm.valuesFileRefs: ConfigMap \"helm-watch-config-map\" not found")
+
+	cmName := "helm-watch-config-map"
+	nt.T.Log("Apply valid ConfigMap that is not immutable (which should not be allowed)")
+	f := false
+	cm0 := fake.ConfigMapObject(core.Name(cmName), core.Namespace(configsync.ControllerNamespace))
+	cm0.Immutable = &f
+	cm0.Data = map[string]string{"something-else.yaml": `
+image:
+  digest: sha256:362cb642db481ebf6f14eb0244fbfb17d531a84ecfe099cd3bba6810db56694e
+  pullPolicy: Never
+`,
+	}
+	nt.T.Cleanup(func() {
+		if err := nt.KubeClient.Delete(cm0); err != nil {
+			nt.T.Log(err)
+		}
+	})
+	if err := nt.KubeClient.Create(cm0); err != nil {
 		nt.T.Fatal(err)
 	}
-	var expectedCPURequest string
-	var expectedCPULimit string
-	var expectedMemoryRequest string
-	var expectedMemoryLimit string
-	if nt.IsGKEAutopilot {
-		expectedCPURequest = "250m"
-		expectedCPULimit = expectedCPURequest
-		expectedMemoryRequest = "512Mi"
-		expectedMemoryLimit = expectedMemoryRequest
-	} else {
-		expectedCPURequest = "150m"
-		expectedCPULimit = "1"
-		expectedMemoryRequest = "250Mi"
-		expectedMemoryLimit = "300Mi"
+	nt.WaitForRootSyncStalledError(rs.Namespace, rs.Name, "Validation", "KNV1061: RootSyncs must only reference ConfigMaps that are immutable; ConfigMap \"helm-watch-config-map\" in namespace \"config-management-system\" is not immutable")
```

Comment:
Can we make this more consistent with the other error message and say "RootSyncs must reference valid ConfigMaps in spec.helm.valuesFileRefs: ConfigMap \"helm-watch-config-map\" in namespace \"config-management-system\" is not immutable"

Reply:
done